### ISSUE TITLE
sphinx-build: usage of `-D` with list of tuples

### ISSUE THREAD

**Issue by diego-plan9** [sphinx-doc/sphinx#2977]
Hello,

I'm wondering if it is possible to use the `-D setting=value` command line argument of `sphinx-build` to override a setting that is defined as a _list of tuples_, for example the `man_pages` of the manual page builder?

I have started trying a naive approach (passing a list separated by commands, as per the documentation for regular lists introduced on 1.3):

```
$ sphinx-build -N -bman -P -D man_pages=source,name,description,author,section docs/ /tmp
Running Sphinx v1.4.6
WARNING: The config value `man_pages' has type `unicode', defaults to `list.'
loading pickled environment... not yet created
building [mo]: targets for 0 po files that are out of date
building [man]: all manpages
updating environment: 5 added, 0 changed, 0 removed
reading sources... [ 20%] changelog
reading sources... [ 40%] comparison
reading sources... [ 60%] index
reading sources... [ 80%] phonetic
reading sources... [100%] stemming

looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... Exception occurred while building, starting debugger:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/sphinx/cmdline.py", line 244, in main
    app.build(opts.force_all, filenames)
  File "/usr/lib/python2.7/dist-packages/sphinx/application.py", line 297, in build
    self.builder.build_update()
  File "/usr/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 246, in build_update
    self.build(['__all__'], to_build)
  File "/usr/lib/python2.7/dist-packages/sphinx/builders/__init__.py", line 322, in build
    self.write(docnames, list(updated_docnames), method)
  File "/usr/lib/python2.7/dist-packages/sphinx/builders/manpage.py", line 57, in write
    docname, name, description, authors, section = info
ValueError: need more than 1 value to unpack
> /usr/lib/python2.7/dist-packages/sphinx/builders/manpage.py(57)write()
-> docname, name, description, authors, section = info
(Pdb) info
u's'
(Pdb) q
```

And went on trying different variations for the `-D man_pages=...` part, with no success.

Could you point me towards the right syntax, if at all possible, or any information about if it is possible to achieve this? I tried to find a related issue or documentation but couldn't locate it - my apologies if this has been already mentioned elsewhere.

Thanks in advance!

```
$ sphinx-build --version
Sphinx (sphinx-build) 1.4.6
```

**Comment by tk0miya** [comment]
AFAIK Sphinx could not accept list of tuples through `-D` option.
Sphinx only supports string, list and integer types.
https://github.com/sphinx-doc/sphinx/blob/1.4.6/sphinx/config.py#L354

We need to improve this conversion to support list of tuples.

**Comment by diego-plan9** [comment]
Thanks for the quick feedback, tk0miya! I'm ok with the alternative option of modifying the settings directly on `conf.py` in the meantime, but thanks for the information and looking into the issue.

**Comment by diego-plan9** [comment]
After looking a bit more into the issue, I'm wondering if a solution based on [ast.literal_eval()](https://docs.python.org/2/library/ast.html#ast.literal_eval) would be acceptable?

In particular, I'm playing with the idea of replacing the [`value.split(',')`](https://github.com/sphinx-doc/sphinx/blob/1.4.6/sphinx/config.py#L364) inside `Config.convert_overrides()`, which in turn would allow the user to pass Python-like constructs to the command line  for settings that expect list of tuples (or other relatively complex types):

``` bash
$ sphinx-build -D somesetting='[("a", "b"), ("e", "f")]' ...
```

This would have also the nice side effects of:
- allowing the passing of lists that have items containing commas (`['a setting, with comma', 'another setting']`), which I _believe_ are currently not properly handled by `split(',')`.
- allow overriding settings with a slightly complex structure such as the [authors for man_page](http://www.sphinx-doc.org/en/stable/config.html#confval-man_pages), which can be a _list within a tuple within a list_ - it might be quite an edge case, but there might be other settings that can contain _lists of lists_, etc. 

As for the drawbacks, I'm concerned about the potential of breaking the current list-passing syntax (and existing setups that rely on it!), but I _believe_ (ie. still need to look in detail and for edge cases) this can be overcome by falling back to the `split(',')` method if the `literal_eval()` call fails, as the latter would fail on unquoted strings. I'm also unsure if allowing the user to pass Python constructs via command line would be considered acceptable, even if `literal_eval()` seems pretty restrictive.

In short: would this be an approach worth exploring? If so,  I'd be happy to start a pull request, try to provide an initial basic version along with tests, and look into the finer details - please be aware that this is basically a result of my first, and very basic, glance at sphinx internals.
